build/net/src/lib.rs

```diff
@@ -82,23 +90,65 @@ pub fn load_net_config() -> Result<NetConfig> {
     Ok(cfg)
 }
 
+pub fn generate_port_consts(
+    config: &NetConfig,
+    mut out: impl std::io::Write,
+) -> Result<(), std::io::Error> {
+    let ports = config
+        .vlans
+        .iter()
+        .map(|v| v.port)
+        .collect::<HashSet<_>>()
+        .len();
+    assert!(ports <= 2);
+    writeln!(out, "pub const PORT_COUNT: usize = {};", ports.max(1))
+}
+
 pub fn generate_vlan_consts(
     config: &NetConfig,
     mut out: impl std::io::Write,
 ) -> Result<(), std::io::Error> {
-    let vlan = config.vlan.unwrap();
-    let end = vlan.start + vlan.count;
-    if end > 0xFFF {
-        panic!("Invalid VLAN range (must be < 4096)");
+    if config.vlans.iter().any(|v| v.vid > 0xFFF) {
+        panic!("Invalid VLAN VID (must be < 4096)");
     }
     writeln!(
         out,
         "
-pub const VLAN_RANGE: core::ops::Range<u16> = {:#x}..{:#x};
-pub const VLAN_COUNT: usize = {};
-",
-        vlan.start, end, vlan.count
-    )
+    pub const VLAN_COUNT: usize = {0};
```

Comment:
It'd be really nice to use quote here instead of the dedented string literals. Not mandatory.

Reply:
Also done in the Grand VLAN refactoring!